docs/source/loading.rst

```diff
@@ -0,0 +1,304 @@
+
+.. code-block::
+
+    # 18 records, from 450 (included) to 468 (excluded).
+    >>> train_50_52pct1_ds = datasets.load_dataset('bookcorpus', split=datasets.ReadInstruction( 'train', from_=50, to=52, unit='%', rounding='pct1_dropremainder'))
+    # 18 records, from 468 (included) to 486 (excluded).
+    >>> train_52_54pct1_ds = datasets.load_dataset('bookcorpus', split=datasets.ReadInstruction('train',from_=52, to=54, unit='%', rounding='pct1_dropremainder'))
+    # Or equivalently:
+    >>> train_50_52pct1_ds = datasets.load_dataset('bookcorpus', split='train[50%:52%](pct1_dropremainder)')
+    >>> train_52_54pct1_ds = datasets.load_dataset('bookcorpus', split='train[52%:54%](pct1_dropremainder)')
+
+.. important::
+
+    Using ``pct1_dropremainder`` rounding may truncate the last examples in a dataset if the number of examples in your dataset don't divide evenly by 100.
+
+Troubleshooting
+---------------
+
+Sometimes, loading a dataset is not as simple as calling :func:`dataset.load_dataset`. In this section, you will learn how to solve two of the most common issues you may encounter when you load a dataset: manually download a dataset, and specify features of a dataset.
+
+Manual download
+^^^^^^^^^^^^^^^
+
+Certain datasets will require you to manually download the dataset files due to licensing incompatibility or if the files are hidden behind a login page. This will cause :func:`dataset.load_dataset` to throw an ``AssertionError``. But Datasets provides detailed instructions for downloading the missing files. After you have downloaded the files, use the ``data_dir`` argument to specify the path to the files you just downloaded. For example, if you try to download the PAN-X.fr configuration from the `xtreme dataset <https://sites.research.google/xtreme>`_:
+
+    >>> dataset = load_dataset("xtreme", "PAN-X.fr")
+    Downloading and preparing dataset xtreme/PAN-X.fr (download: Unknown size, generated: 5.80 MiB, total: 5.80 MiB) to /Users/thomwolf/.cache/huggingface/datasets/xtreme/PAN-X.fr/1.0.0...
+    AssertionError: The dataset xtreme with config PAN-X.fr requires manual data.
+    Please follow the manual download instructions: You need to manually download the AmazonPhotos.zip file on Amazon Cloud Drive (https://www.amazon.com/clouddrive/share/d3KGCRCIYwhKJF0H3eWA26hjg2ZCRhjpEQtDL70FSBN). The folder containing the saved file can be used to load the dataset via 'datasets.load_dataset("xtreme", data_dir="<path/to/folder>")'
+
+Specify features
+^^^^^^^^^^^^^^^^
+
+When you create a dataset from local files, the :class:`dataset.Features` are automatically generated by `Apache Arrow's Automatic Type Inference <https://arrow.apache.org/docs/python/json.html#automatic-type-inference>`_. However, the features of the dataset may not always align with your expectations or you may want to define the features yourself. 
```

Comment:
This link to the Apache Arrow's website is for JSON, not sure this is relevant here

Reply:
Maybe this [link](https://arrow.apache.org/docs/python/data.html) is more appropriate for describing Arrow's automatic type inference?